### PR TITLE
[TA-1235] Refactor signer to use deep links

### PR DESCRIPTION
**Pull request by GuillemGarciaDev**
# [TA-1235] Refactor signer to use deep links

## Tasks

- [[TA-1235] Refactor signer to use deep links](https://www.notion.so/Refactor-signer-to-use-deep-links-6868614f346f40178421d52d0e379e4e?pvs=4)

## Dependencies

NearMobileSigner: [[TA-1235] Refactor signer to use deep links](https://github.com/Peersyst/near-mobile-signer/compare/feat/deep-linking?expand=1)

## Changes

- `Navigator.linking` containing all linking settings
- updated `Navigator` to accept a `linking` config
- `useShowSignerRequest` hook to handle signer request deep links
- Merged `MainBottomScreens` into `MainScreens` to handle deep link path
- updated QR code handling
- updated locale

## Notes

THIS TASK IS NOT FINISHED. A PR in [near-mobile-signer](https://github.com/Peersyst/near-mobile-signer/compare/feat/deep-linking?expand=1) package must be merged before merging this one.

### CODE REVIEW

File: src/module/signer/hooks/useShowSignerRequest.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { useRoute } from "@react-navigation/native";
+import useSignerModal, { SignerModalType } from "./useSignerModal";
+import useNavigation from "module/common/hook/useNavigation";
+import { MainScreens } from "module/common/component/navigation/MainNavigatorGroup/MainScreens";
+import { useCallback } from "react";
+
+export default function useShowSignerRequest() {
+    const { path, params } = useRoute();
```
[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/module/signer/hooks/useShowSignerRequest.ts:8]: yes!

[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/module/signer/hooks/useShowSignerRequest.ts:8]: Per haver-ho declarat al linking com `sign/:type/:id`

File: src/module/home/screen/HomeScreen.tsx [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
 import MainGradientScreen from "module/main/component/layout/MainGradientScreen/MainGradientScreen";
+import useShowSignerRequest from "module/signer/hooks/useShowSignerRequest";
+import { useAppState } from "@react-native-community/hooks";
+import { useEffect } from "react";
 
 const HomeScreen = (): JSX.Element => {
+    const appState = useAppState();
+    const showSignerRequest = useShowSignerRequest();
+
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/module/home/screen/HomeScreen.tsx:12]: En veritat no cal, perque cada cop que es cridi el deeplink entrara a home

[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/module/home/screen/HomeScreen.tsx:12]: Ho cridaries a pelo a la `HomeScreen` llavors?

[the remark this answers is not in the document]
Comment (AgustinMJ) [reply src/module/home/screen/HomeScreen.tsx:12]: Recordatori de treure-ho

**Pull request merged**